lib/spack/spack/installer.py

```diff
@@ -0,0 +1,1554 @@
+                lock = None
+            except (Exception, KeyboardInterrupt, SystemExit) as exc:
+                tty.error(err.format(res, pkg_id, exc.__class__.__name__,
+                          str(exc)))
+                self._cleanup_all_tasks()
+                raise
+
+            self.locks[pkg_id] = ('write', lock)
+
+        return 'write', lock
+
+    def _init_queue(self, install_deps, install_self):
+        """
+        Initialize the build task priority queue and spec state.
+
+        Args:
+            install_deps (bool): ``True`` if installing package dependencies,
+                otherwise ``False``
+            install_self (bool): ``True`` if installing the package, otherwise
+                ``False``
+        """
+        tty.debug('Initializing the build queue for {0}'.format(self.pkg.name))
+        install_compilers = spack.config.get(
+            'config:install_missing_compilers', False)
+
+        if install_deps:
+            for dep in self.spec.traverse(order='post', root=False):
+                dep_pkg = dep.package
+
+                # First push any missing compilers (if requested)
+                if install_compilers:
+                    self._add_bootstrap_compilers(self, dep_pkg)
+
+                if dep_pkg.unique_id not in self.build_tasks:
+                    self._push_task(dep_pkg, False, 0, 0, STATUS_ADDED)
+
+                # Clear any persistent failure markings _unless_ they are
+                # associated with another process in this parallel build
+                # of the spec.
+                spack.store.db.clear_failure(dep, force=False)
+
+            # Push any missing compilers (if requested) as part of the
+            # package dependencies.
+            if install_compilers:
+                self._add_bootstrap_compilers(self, self.pkg)
+
+        if install_self and self.pkg.unique_id not in self.build_tasks:
+            # Now add the package itself, if appropriate
+            self._push_task(self.pkg, False, 0, 0, STATUS_ADDED)
+
+    def _install_task(self, task, **kwargs):
+        """
+        Perform the installation of the requested spec and or dependency
+        represented by the build task.
+
+        Args:
+            task (BuildTask): the installation build task for a package"""
+
+        cache_only = kwargs.get('cache_only', False)
+        dirty = kwargs.get('dirty', False)
+        fake = kwargs.get('fake', False)
+        install_source = kwargs.get('install_source', False)
+        keep_stage = kwargs.get('keep_stage', False)
+        skip_patch = kwargs.get('skip_patch', False)
+        tests = kwargs.get('tests', False)
+        use_cache = kwargs.get('use_cache', True)
+        verbose = kwargs.get('verbose', False)
+
+        pkg = task.pkg
+        explicit = pkg.unique_id == self.pkg.unique_id
+
+        tty.msg(get_install_msg(pkg.unique_id))
+        task.start = task.start if task.start else time.time()
```

Comment:
```suggestion
        task.start = task.start or time.time()
```

Reply:
Done.

Reply:
Hm.. I thought I changed this but it isn't showing as "Outdated".